### ISSUE TITLE
[TextField + Icon] Leading icon has incorrect padding

### ISSUE THREAD

**Issue by youra-h** [material-components/material-components-web#6618]
### Steps to reproduce
Taken from discord
![image](https://user-images.githubusercontent.com/5162766/99177795-95cab200-272e-11eb-9d48-12ad62a70ae4.png)
For icon button necessary to use mdc-text-field__icon--trailing.margin-right: 4px;
Stille like this for prefix and suffix
```scss
/* TEXTFIELD PREFIX SUFFIX */
.mdc-text-field__affix--prefix {
    padding-right: 4px;
}
.mdc-text-field__affix--suffix {
    padding-left: 4px;
}
```

And
![image](https://user-images.githubusercontent.com/5162766/99177852-243f3380-272f-11eb-98dd-e5436351a652.png)

![image](https://user-images.githubusercontent.com/5162766/99177813-ce6a8b80-272e-11eb-9891-e763babaccae.png)

Final code
```scss
/* TEXTFIELD HELPER LINE */
.mdc-text-field+.mdc-text-field-helper-line {
    justify-content: left;
    padding: 0 14px;
}

/* TEXTFIELD ICONS */

.mdc-text-field {
    .mdc-icon-button {
        @include icon-button.size(44px);
    }
}

.mdc-text-field__icon--leading,
.mdc-text-field__icon--trailing {
    padding: 10px;
}

.mdc-text-field__icon--leading {
    margin: 0 4px;
}

.mdc-text-field__icon--trailing {
    margin: 0 4px;
}

.mdc-text-field--with-leading-icon {
    &.mdc-text-field--filled {
        .mdc-floating-label {
            left: 52px;
        }
    }
}

/* TEXTFIELD PREFIX SUFFIX */
.mdc-text-field__affix--prefix {
    padding-right: 4px;
}
.mdc-text-field__affix--suffix {
    padding-left: 4px;
}
```

### Your Environment:

| Software         | Version(s) |
| ---------------- | ---------- |
| MDC Web          |8.0.0
| Browser          |Opera/Chrome
| Operating System |Windows

**Comment by devslyfox** [comment]
```scss
/* TEXTFIELD */
.mdc-text-field-helper-line,
.mdc-text-field {
    --padding-text-field: 14px;
}

.mdc-text-field {
    padding: 0 var(--padding-text-field, 16px);
}

/* TEXTFIELD LABEL */
.mdc-text-field--filled {
    .mdc-floating-label {
        left: var(--padding-text-field, 16px);
    }
}


/* TEXTFIELD HELPER LINE */
.mdc-text-field+.mdc-text-field-helper-line {
    justify-content: left;
    padding: 0 var(--padding-text-field, 16px);
}

/* TEXTFIELD ICONS */

.mdc-text-field--with-trailing-icon {
    padding-left: var(--padding-text-field, 16px);
    padding-right: 0;
}

.mdc-text-field--with-leading-icon {
    padding-right: var(--padding-text-field, 16px);
    padding-left: 0;
    &.mdc-text-field--filled {
        // GOTO here you have to calc
        .mdc-floating-label {
            left: 52px;
        }
    }
}

// GOTO here you have to calc
.mdc-text-field {
    .mdc-icon-button {
        @include icon-button.size(48px);
    }
}

.mdc-text-field__icon--leading,
.mdc-text-field__icon--trailing {
    padding: 10px;
}

.mdc-text-field__icon--leading {
    margin: 0 2px 0 4px;
}

.mdc-text-field__icon--trailing {
    margin: 0 4px 0 2px;
}

/* TEXTFIELD PREFIX SUFFIX */
.mdc-text-field__affix--prefix {
    padding-right: 4px;
}
.mdc-text-field__affix--suffix {
    padding-left: 4px;
}

```

**Comment by asyncLiz** [comment]
@sayris could you take a look into this if you have time? I believe you fixed the trailing icon touch target and I think the leading icon needs fixed as well.

**Comment by sayris** [comment]
Hi @devslyfox -- thanks for flagging this issue!

Can you confirm that this bug is about the discrepancies in padding between the leading and trailing icon in the textfield (and so a fix would increase the padding on the leading icon to match that of trailing icon)? Unfortunately, I can't discern what is adjusted in the screenshots following the "I think this is how it should be" message, so want to make sure I understand the issue fully before working on a fix based on my assumptions :) 

Thanks!

**Comment by devslyfox** [comment]
Look, this post is not related to the current question. This is a discord picture from some user. But it also shows that the icon goes beyond the boundaries.
![image](https://user-images.githubusercontent.com/5162766/99424759-e1927c80-2923-11eb-8408-a2a1fc22d000.png)

I did like this:
Leading icon
![image](https://user-images.githubusercontent.com/5162766/99426069-7d70b800-2925-11eb-88df-6a0ac3dc9408.png)
![image](https://user-images.githubusercontent.com/5162766/99426146-937e7880-2925-11eb-86bb-7ef19d1356b8.png)
Leading button
![image](https://user-images.githubusercontent.com/5162766/99426347-ca548e80-2925-11eb-98b8-3675d01f7221.png)

Trailing icon
![image](https://user-images.githubusercontent.com/5162766/99426421-dd675e80-2925-11eb-8044-ec471213a5ed.png)
![image](https://user-images.githubusercontent.com/5162766/99426450-e8ba8a00-2925-11eb-9305-7a820e6179dd.png)
Trailing button
![image](https://user-images.githubusercontent.com/5162766/99426497-f839d300-2925-11eb-8140-31141bd960e3.png)

**Comment by asyncLiz** [comment]
@sayris here's a jsbin showcasing the touch target issue: https://jsbin.com/posiyaqamu/edit?html,output

The leading icon does not have a 48x48 touch target while the trailing icon does, due to it using margin instead of padding.

![Screen Shot 2020-11-17 at 11 06 12 AM](https://user-images.githubusercontent.com/34382620/99435407-eecc4e80-28c4-11eb-9eba-814a89b99061.png)
![Screen Shot 2020-11-17 at 11 06 15 AM](https://user-images.githubusercontent.com/34382620/99435415-f1c73f00-28c4-11eb-975d-ad1233f2c40a.png)

@devslyfox the icon's touch target should meet the edge of the text field, there should not be any additional margin added. If a 48x48 ripple is added, that will result in the ripple meeting the edge of the text field container as well. You can reduce the size of the ripple if this is undesirable, or add your own margin if you wish the ripple to remain the same size.

At the moment we don't have explicit guidance from design on what the icon's ripple should look like for text field.

**Comment by wouterSkepp** [comment]
> padding

I'd like to chime in that the *icon--leading and *icon--trailing do not have a 48x48px touch target - It is a side-effect of using material-icon. If one would swap out the iconset for font-awesome (for example), it is not enforced, offsetting the floating label.

I would suggest adding the sizing to the relevant `.mdc-text-field__icon`
This floating-label offset is also happening in the select component with `.mdc-select__icon`.
